src/java/htsjdk/samtools/sra/SRALazyRecord.java

```diff
@@ -0,0 +1,941 @@
+                    method.setAccessible(true); // Impl method is private
+                    ret = method.invoke(this);
+                } catch (IllegalAccessException e) {
+                    throw new RuntimeException(e);
+                } catch (NoSuchMethodException e) {
+                    throw new RuntimeException(e);
+                } catch (InvocationTargetException e) {
+                    throw new RuntimeException(e);
+                }
+                setAttribute(tag, ret);
+            }
+        }
+        return super.getAttribute(tag);
+    }
+
+    @Override
+    public void setAttribute(final short tag, final Object value) {
+        AttributeTagInfo info = lazyAttributeTags.get(tag);
+        if (info != null && !initializedAttributes.contains(info.lazyAttribute)) {
+            initializedAttributes.add(info.lazyAttribute);
+        }
+        super.setAttribute(tag, value);
+    }
+
+    @Override
+    protected void setAttribute(final short tag, final Object value, final boolean isUnsignedArray) {
+        AttributeTagInfo info = lazyAttributeTags.get(tag);
+        if (info != null && !initializedAttributes.contains(info.lazyAttribute)) {
+            initializedAttributes.add(info.lazyAttribute);
+        }
+        super.setAttribute(tag, value, isUnsignedArray);
+    }
+
+    @Override
+    public void clearAttributes() {
+        for (LazyAttribute lazyAttribute : LazyAttribute.values()) {
+            if (!initializedAttributes.contains(lazyAttribute)) {
+                initializedAttributes.add(lazyAttribute);
+            }
+        }
+        super.clearAttributes();
+    }
+
+    @Override
+    protected void setAttributes(final SAMBinaryTagAndValue attributes) {
+        for (LazyAttribute lazyAttribute : LazyAttribute.values()) {
+            if (!initializedAttributes.contains(lazyAttribute)) {
+                initializedAttributes.add(lazyAttribute);
+            }
+        }
+        super.setAttributes(attributes);
+    }
+
+    @Override
+    protected SAMBinaryTagAndValue getBinaryAttributes() {
+        for (Map.Entry<Short, AttributeTagInfo> info : lazyAttributeTags.entrySet()) {
+            if (!initializedAttributes.contains(info.getValue().lazyAttribute)) {
+                getAttribute(info.getKey());
+            }
+        }
+
+        return super.getBinaryAttributes();
+    }
+
+    public boolean isUnsignedArrayAttribute(final String tag) {
+        Short binaryTag = SAMTagUtil.getSingleton().makeBinaryTag(tag);
+        AttributeTagInfo info = lazyAttributeTags.get(binaryTag);
+        if (info != null && !initializedAttributes.contains(info.lazyAttribute)) {
+            getAttribute(binaryTag);
+        }
+
+        return super.isUnsignedArrayAttribute(tag);
+    }
+
+    // ===== misc ====
+
+    @Override
+    public boolean equals(final Object o) {
+        throw new RuntimeException("comparing of SRALazyRecords is not implemented yet");
```

Comment:
You need to implement `equals()`, `hashCode()`, and `clone()`, otherwise we will not be able to use this `SAMRecord` implementation correctly in the GATK.